src/include/gurt/common.h

```diff
@@ -574,12 +574,37 @@ d_sgl_buf_copy(d_sg_list_t *dst_sgl, d_sg_list_t *src_sgl)
 	}
 }
 
-void d_getenv_bool(const char *env, bool *bool_val);
-void d_getenv_char(const char *env, char *char_val);
-void d_getenv_int(const char *env, unsigned int *int_val);
-int d_getenv_uint64_t(const char *env, uint64_t *val);
-int  d_write_string_buffer(struct d_string_buffer_t *buf, const char *fmt, ...);
-void d_free_string(struct d_string_buffer_t *buf);
+bool
+d_isenv_def(char *name);
+int
+d_getenv_str(char *str_val, size_t str_size, const char *name);
+int
+d_agetenv_str(char **str_val, const char *name);
+void
+d_free_env(char **str_val);
+int
+d_getenv_bool(bool *bool_val, const char *env);
```

Comment:
This is changing the API and ABI isn't it?  Existing applications which use this interface will need code changes to work, but if they build with one version and run with another they could also segfault.

Reply:
I had a similar thought but then kind of assumed that the only user of these APIs are DAOS libraries that will be shipped with the changes.

Reply:
At this time, the only user of this new APIs are indeed the DAOS libraries and thus should not have any side effects for the client applications linked with DAOS.  They will just use the standard non-thread safe `getenv()` functions.